### PR TITLE
Fix Encoding.CodePages test on desktop

### LINKED ISSUE #10886: Test failure: System.Text.Tests.EncodingTest/TestDefaultEncodings
Opened on behalf of @jiangzeng

The test `System.Text.Tests.EncodingTest/TestDefaultEncodings` has failed.

```
Assert.Throws() Failure\r
Expected: typeof(System.NotSupportedException)\r
Actual:   (No exception was thrown)
```

Stack Trace:

```
   at System.Text.Tests.EncodingTest.TestDefaultEncodings() in D:\A\_work\2\s\src\System.Text.Encoding.CodePages\tests\EncodingCodePages.cs:line 464
```

Failing configurations:
- Windows.10.Amd64
  - AnyCPU-Release
  - AnyCPU-Debug

Details:
https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster/type/test~2Ffunctional~2Fuwp~2F/build/20160816.08/workItem/System.Text.Encoding.CodePages.Tests/analysis/xunit/System.Text.Tests.EncodingTest~2FTestDefaultEncodings

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes #10886

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @danmosemsft

File: src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs [hunk 1]
```diff
@@ -458,20 +458,28 @@ public static void TestDefaultEncodings()
...
             // The default encoding should be something from the known list.
             Encoding defaultEncoding = Encoding.GetEncoding(0);
             Assert.NotNull(defaultEncoding);
             KeyValuePair<int, string> mappedEncoding = Map(defaultEncoding.CodePage, defaultEncoding.WebName);
-            Assert.Contains(mappedEncoding, CrossplatformDefaultEncodings());
+
+            if (defaultEncoding.CodePage == Encoding.UTF8.CodePage)
+            {
+                // if the default encoding is not UTF8 that means either we are running on the full framework
+                // or the provider is registered. at that time we shouldn't expect exceptions when creating
```
Comment (danmosemsft) [inline src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs]: What provider? Do you mean 'not registered'?

File: src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs [hunk 2]
```diff
@@ -458,20 +458,28 @@ public static void TestDefaultEncodings()
...
+                // the following encodings.
+                foreach (object[] mapping in CodePageInfo())
+                {
+                    Assert.Throws<NotSupportedException>(() => Encoding.GetEncoding((int)mapping[0]));
+                    Assert.Throws<ArgumentException>(() => Encoding.GetEncoding((string)mapping[2]));
+                }
+
+                // Currently the class EncodingInfo isn't present in corefx, so this checks none of the code pages are present.
+                // When it is, comment out this line and remove the previous foreach/assert.
+                // Assert.Equal(CrossplatformDefaultEncodings, Encoding.GetEncodings().OrderBy(i => i.CodePage).Select(i => Map(i.CodePage, i.WebName)));
```
Comment (danmosemsft) [inline src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs]: Cite issue ID maybe?

File: src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs]: Also one might expect you mean 'not registered' if it means you stop getting exceptions, but I don't know how this provider works.

Comment (tarekgh) [inline src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs]: I'll modify the comment to be clearer. thanks

---
In reply to: [95671559](https://github.com/dotnet/corefx/pull/15079#discussion_r95671559) [](ancestors = 95671559)

**Pull request merged**